### DISCUSSION TITLE
[BP#69] Ne pas appeler de fonction dans la déclaration d’une boucle de type for

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#219] [Bonnes pratiques] [answered]
C'est encore d'actualité en PHP? Ça s'applique à d'autres languages? Ce n'est pas optimiser à la compilation (Inc. à la volée) ?

Peut-être à tester pour voir réellement l'impact. J'ose espérer que l'exemple donné à un effet proche de 0, peut-être mettre un exemple plus probant ?

**Comment by ACTLEM** [comment, marked as answer]
Déjà quelque soit le langage je ne suis pas fan de cette pratique (mais cela relève plus de la qualité logiciel et non de l'écoconception). Après, intéressé d'avoir des avis sur l'effet réel en PHP.

**Reply by ACTLEM** [reply]
@arnolem peux-tu nous éclairer sur l'intérêt ou pas de cette BP stp?

**Reply by ACTLEM** [reply]
Etant donné l'absence de preuve pour l'instant sur l'intérêt d'un point de vue écoconception (même si côté qualité logiciel cela peut s'entendre), je partirais pour la supprimer. D'autres avis?

**Reply by rlemaire** [reply]
+1 pour supprimer.

**Reply by ACTLEM** [reply]
Issue pour suppression: https://github.com/cnumr/best-practices/issues/306
